cmd/juju/application/updateseries.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2017 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+
+	"github.com/juju/cmd"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"gopkg.in/juju/charm.v6-unstable"
+	"gopkg.in/juju/charmrepo.v2-unstable"
+	"gopkg.in/juju/names.v2"
+	"gopkg.in/macaroon-bakery.v1/httpbakery"
+
+	"github.com/juju/juju/api"
+	"github.com/juju/juju/api/application"
+	"github.com/juju/juju/api/modelconfig"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/cmd/juju/block"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/environs/config"
+)
+
+// NewUpdateSeriesCommand returns a command which updates the series of
+// an application or machine.
+func NewUpdateSeriesCommand() cmd.Command {
+	cmd := &updateSeriesCommand{
+		NewUpdateApplicationClient: func(conn api.Connection) UpdateApplicationClient {
+			return application.NewClient(conn)
+		},
+		//NewUpdateMachineClient: func(conn api.Connection) UpdateMachineClient {
+		//	return machinemanager.NewClient(conn)
+		//},
+		NewModelConfigGetter: func(conn api.Connection) ModelConfigGetter {
+			return modelconfig.NewClient(conn)
+		},
+	}
+	return modelcmd.Wrap(cmd)
+}
+
+// UpdateApplicationClient defines a subset of the application facade, as required
+// by the update-series command.
+type UpdateApplicationClient interface {
+	Get(string) (*params.ApplicationGetResults, error)
+	GetCharmURL(string) (*charm.URL, error)
+	UpdateApplicationSeries(series params.ApplicationUpdateSeries) error
+}
+
+// UpdateMachineClient defines a subset of the machine facade, as required by
+// update-series command.
+/*
+type UpdateMachineClient interface {
+	UpdateMachineSeries()
+}
+*/
+
+// UpdateSeries is responsible for updating the series of an application or machine.
+type updateSeriesCommand struct {
+	modelcmd.ModelCommandBase
+
+	NewUpdateApplicationClient func(connection api.Connection) UpdateApplicationClient
+	//NewUpdateMachineClient     func(connection api.Connection) UpdateMachineClient
+	NewModelConfigGetter func(api.Connection) ModelConfigGetter
+
+	ApplicationName string
+	Force           bool
+	MachineNumber   string
+	Series          string
+}
+
+var updateSeriesDoc = `
+It is recommended to update the series on existing units before running this
+command.
+
+By updating the application series, future units of the application to use
+the new series.  The series for subordinate applications is also updated.
+
+The update is disallowed unless the --force flag is used if the series is not
+explicitly supported:
```

Comment:
"... if the requested series is not explicitly supported by the application's charm and any subordinates, as well as any other charms which may be deployed to the same machine." 

You can add a comment that the user may wish to upgrade the charm to a later revision that does support their preferred series.